pkgs/development/compilers/dotnet/sdk/default.nix

```diff
@@ -0,0 +1,280 @@
+  '';
+
+    installPhase = ''
+      cp -Tr . "$out"
+    '';
+    postFixup = ''
+      for p in $(find "$out" -type f -executable); do
+        if isELF "$p"; then
+          echo "Patchelfing $p"
+          patchelf \
+            --set-interpreter "${stdenv.cc.bintools.dynamicLinker}" \
+            "$p" ||:
+          patchelf \
+            --set-rpath "${binary-rpath}" \
+            "$p" ||:
+        fi
+      done
+    '';
+
+    passthru.packages = let
+      artifacts = pkgs.fetchurl {
+        url = "https://dotnetcli.azureedge.net/source-built-artifacts/assets/Private.SourceBuilt.Artifacts.6.0.107.tar.gz";
+        sha256 = "14mzn2wy6ny5n94f93gdbm0p7scx6qjph8n3bm2p8vnwkhbq35gw";
+      };
+
+      in stdenv.mkDerivation {
+        name = artifacts.name;
+
+        src = artifacts;
+
+        sourceRoot = ".";
+
+        nativeBuildInputs = [ pkgs.unzip pkgs.zip ];
+
+    postPatch = ''
+      for x in *.linux-x64.*.nupkg
+      do
+        mkdir .patch
+        pushd .patch
+        unzip -q ../"$x"
+        chmod -R +rw .
+        for p in $(find -type f); do
+          if isELF "$p"; then
+            echo "Patchelfing $p"
+            patchelf \
+              --set-interpreter "${stdenv.cc.bintools.dynamicLinker}" \
+              "$p" ||:
+            patchelf \
+              --set-rpath "${binary-rpath}" \
+              "$p" ||:
+          fi
+        done
+        # Sets all timestamps to Jan 1 1980, the earliest mtime zips support.
+        find -exec touch -t 198001010000.00 {} +
+        zip -r ../"$x" .
+        popd
+        rm -rf .patch
+      done
+    '';
+
+        installPhase = ''
+          cp -Tr . $out
+        '';
+      };
+  };
+
+  dotnet-sdk-src = with pkgs; runCommand "dotnet-build-src" ({
+    outputHashAlgo = "sha256";
+    outputHash = "sha256-xnRd0ctsDrWGKca1xCArUz7J16P48Awnc7T3r6BnP+c=";
+    outputHashMode = "recursive";
+    nativeBuildInputs = [
+      wget
+      cacert
+      git
+      jq
+    ];
+    src = fetchFromGitHub {
+      owner = "dotnet";
+      repo = "installer";
+      rev = "refs/tags/v6.0.108";
+      sha256 = "sha256-ssGp9OiunlLXuVw1kpSxYzh7GbkO0AaMpDPqVZFyw+U=";
+      leaveDotGit = true;
+    };
+    DOTNET_INSTALL_DIR = bootstrap-sdk;
+    # https://github.com/NixOS/nixpkgs/issues/38991
+    # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
+    LOCALE_ARCHIVE = lib.optionalString stdenv.isLinux
+        "${glibcLocales}/lib/locale/locale-archive";
+  }) ''
+    cp -Tr "$src" src
+    chmod -R +w src
+    cd src
+    git remote add origin https://github.com/dotnet/installer.git
+    patch -p1 < ${./0001-make-tarball-deterministic.patch}
+    jq '.tools=(.tools|.dotnet="6.0.107"|del(.runtimes))' global.json > global.json~
+    mv global.json~ global.json
+    patchShebangs *.sh
+    HOME=$(pwd)/fake-home \
+    DotNetBuildFromSource=true \
+      ./build.sh /p:ArcadeBuildTarball=true /p:TarballDir=\"$out\" /p:OfficialBuildId=20220101.1
```

Comment:
This is building a source 'tarball' (actually loose files) : https://github.com/dotnet/installer#building, hence the fixed output derivation.

As i mentioned earlier, I think long-term, we'll want to try to recreate the tarball ourselves (it's quite huge) in nix, but since this is the documented way to do it (and what other distros do), I thought it would be best to start with it.

Reply:
Yeah, honestly, for long-term maintainability and ease of use, I think we should stick to the official instructions even if we have the ability to create the tarball ourselves, if anything changes it'll be a royal pain to find out how it is doing things from scratch again.